packages/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -0,0 +1,217 @@
+                    this._unwrap(moduleExports.Server.prototype, 'on');
+                }
+            },
+            [broadcastOperatorInstrumentation, namespaceInstrumentation, socketInstrumentation]
+        );
+    }
+
+    private _patchOn() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, originalListener: Function) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const wrappedListener = function () {
+                    const operation = 'on';
+                    const messageName = ev;
+
+                    const span: Span = self.tracer.startSpan(`socket.io ${operation} ${messageName}`, {
+                        kind: SpanKind.CONSUMER,
+                        attributes: {
+                            [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                            [SemanticAttributes.MESSAGING_DESTINATION]: originalListener.name,
+                            [SemanticAttributes.MESSAGING_OPERATION]: MessagingOperationValues.PROCESS,
+                        },
+                    });
+
+                    if (self._config.onHook) {
+                        safeExecuteInTheMiddle(
+                            () => self._config.onHook(span, { args: arguments }),
+                            (e) => {
+                                if (e) diag.error(`socket.io instrumentation: onHook error`, e);
+                            },
+                            true
+                        );
+                    }
+
+                    const result = context.with(setSpan(context.active(), span), () =>
+                        originalListener.apply(this, arguments)
+                    );
+                    span.end();
+                    return result;
+                };
+
+                return original.apply(this, [ev, wrappedListener]);
+            };
+        };
+    }
+
+    private _patchEmit() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, ...args: any[]) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const operation = 'emit';
+                const messageName = ev;
+                const attributes = {
+                    [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                    [SemanticAttributes.MESSAGING_DESTINATION_KIND]: 'topic',
+                    [SemanticAttributes.MESSAGING_OPERATION]: operation,
+                };
+
+                const rooms = this.rooms || this._rooms || this.sockets._rooms;
+                if (rooms && rooms.size > 0) {
+                    attributes['socket.io.rooms'] = Array.from<string>(rooms).join();
+                }
+
+                if (this.name) {
```

Comment:
this function is called also on `Server.emit`, in which case `this` will point to Server.
IMO, we should check that we are patching broadcast function before adding an attribute with `namespace` key.

Reply:
`Server.emit` has a default namespace of `/` so maybe it makes sense to also include this for all server.emits?

Reply:
removed the patch from `Server.emit` since internally it uses `namespace.emit` and that is already patched